### PR TITLE
This PR adds a feature to find the median of a stream of numbers using a two-heap approach.

### LINKED ISSUE #1886: Find the Median of a Stream (using Two Heaps)
### Feature Name

Add Find the Median of a Stream in Heaps Folder

### Feature Description

To find the median of a stream of numbers, two heaps are used: a Max Heap for the lower half and a Min Heap for the upper half. The two heaps allow constant-time retrieval of the median by balancing the sizes of the heaps. This approach efficiently handles dynamic data insertion.

### Motivation

These heap-based algorithms are essential for beginners to learn DSA, as they provide efficient solutions to common problems involving sorting, priority queues, and dynamic data processing, helping to build a strong foundation in algorithm design and optimization.

### Feature Type

New Algorithm

@ajay-dhangar I would like to work on it.
Please assign it to me.

### PR DESCRIPTION
**Pull request by tatheer-fathima**
### Description
Implemented `MedianFinder` class with `addNum` and `findMedian` methods.
- Used max-heap and min-heap to maintain the balance for efficient median calculation.

Fixes #1886 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/heap/Add-Merge-k-sorted-lists.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+id: Heap-data-Structure-4
```
Comment (ajay-dhangar) [inline docs/heap/Add-Merge-k-sorted-lists.md]: replace `id: Heap-data-Structure-4` with `id: heap-data-structure-4`

File: docs/heap/Median-of-a-stream.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+id: Heap-data-Structure-5
```
Comment (ajay-dhangar) [inline docs/heap/Median-of-a-stream.md]: replace `id: Heap-data-Structure-5` with `id: heap-data-structure-5`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by tatheer-fathima** [comment]
@ajay-dhangar Ok done. Now u can check.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**